interface/ceed-fortran.c

```diff
@@ -102,7 +102,7 @@ void fCeedView(int *ceed, int *err) {
 
 #define fCeedDestroy FORTRAN_NAME(ceeddestroy,CEEDDESTROY)
 void fCeedDestroy(int *ceed, int *err) {
-  if (Ceed_n == 0) return;
+  if (Ceed_n == 0 || !Ceed_dict[*ceed]) return;
```

Comment:
Ah, before this commit I would have had exactly the issue you described.

Reply:
Oh, I see. There are still inconsistencies, but hard to fix unless we're allowed to ask Fortran for an address-sized argument.
```fortran
ceed c1,c2
call ceedinit(resource, c1, err)
call ceeddestroy(c1, err)
call ceedinit(resource, c2, err) ! reuses old index value
call ceeddestroy(c1, err) ! actually destroys c2
```

The type safety is so much better in recent versions of Fortran, it makes me sad we're stuck with this.

Reply:
I was puzzling on how to prevent that case. I think I can add a small fix to prevent that case!